shell/browser/native_window_mac.mm

```diff
@@ -238,6 +239,12 @@ void SetFrameSize(NSView* self, SEL _cmd, NSSize size) {
   super_impl(self, _cmd, size);
 }
 
+// In simpleFullScreen mode, update the frame for new bounds.
+void UpdateFrame(NSWindow* window) {
+  NSRect fullscreenFrame = [window.screen frame];
```

Comment:
Is it safe to access `NSWindow` in the callback, is there a guarantee that it will be alive when called ?

Can the post task call be changed to act on `weakptr` receiver, like

```
base::PostTask(
      FROM_HERE, {content::BrowserThread::UI},
      base::BindOnce(&NativeWindow::UpdateFrame, GetWeakPtr()));
```

Reply:
sure - sounds good to me! i think it should be safe to do as is but minds well be safer